src/api/analytics/gtag.js

```diff
@@ -0,0 +1,192 @@
+import { links, options } from 'config'
+import formatFacetNames from 'helpers/formatFacetNames'
+import formatFilterName from 'helpers/formatFilterName'
+
+const { transformationHumanReadable: transformation } = options
+
+// adds a custom event to GA4
+// https://developers.google.com/analytics/devguides/collection/ga4/events?client_type=gtag
+const event = (eventName, value = {}, nonInteraction = false) => {
+  window.gtag('event', eventName, {
+    ...value,
+    non_interaction: nonInteraction
+  })
+}
+
+/* --- General --- */
+const emailSubscription = (from) =>
+  event('email_subscription', { subscription_click_from: from })
+
+/* --- Compendia --- */
+const compendiaDownload = (type, organism) =>
+  event(`compendia_${type}_download`, {
+    [`${type}_organism`]: organism
+  })
+
+/* --- Datasets --- */
+const myDatasetAction = (action) =>
+  event('my_dataset_action', {
+    my_dataset_action: `${action[0].toUpperCase()}${action.substring(
+      1
+    )} Samples`
+  })
+
+const myDatasetDownloadOptions = (dataset) =>
+  event('my_dataset_download_options', {
+    my_dataset_download_options: `Aggregate: ${
+      dataset.aggregate_by
+    }, Transformation: ${transformation[dataset.scale_by]}, QN: ${
+      dataset.quantile_normalize ? 'Skipped' : 'Not skipped'
+    }
+  `
+  })
+
+const datasetDownload = (token) => event('dataset_downalod', { token })
+
+const oneOffExperimentDownload = (accessionCode) =>
+  event('one_off_experiment_download', {
+    one_off_experiment_download: accessionCode
+  })
+
+const regeneratedDataset = (state, defaultOptions, newOptions) => {
+  // due to GA char limit, keys names are abbreviated
+  const format = (option) => {
+    return `A: ${option.aggregate_by}, T: ${
+      transformation[option.scale_by]
+    }, QN: ${option.quantile_normalize ? 'Not skipped' : 'Skipped'}`
+  }
+
+  event('regenerated_dataset', {
+    regenerated_state: state ? 'Expired' : 'Valid',
+    regenerated_download_options: `${format(defaultOptions)} ${
+      newOptions ? `(new) ${format(newOptions)}` : ''
+    }`
+  })
+}
+
+const sharedDataset = (status) =>
```

Comment:
Is status going to be a boolean?

Reply:
I've renamed this param name to `isProcessed` for clearer indication of its data type at a5d1a15fff0ea82a489f79446cdc8501bbc23f5.

Reply:
I've also renamed other parameters for consistency at 2bb98c8c735e28b65a0f113aa3984fce1af84e56, thank you David!